source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
This doesn't work. While you've made the `counter` variable `atomic` (which is correct and should work), you've now left the other members unprotected. In particular, the lines below access `counter_map`, which is no longer protected.

...though on second though, it seems to me that we don't use `counter_map` in multithreaded mode -- do you agree? If so, maybe put a comment about this somewhere in the next couple of lines?